code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -19,7 +19,22 @@
 	new /obj/effect/temp_visual/telekinesis(get_turf(caller))
 	caller.visible_message("<span class='xenowarning'>A strange buzzing hum starts to emanate from \the [caller]!</span>", \
 	"<span class='xenodanger'>We call forth the larvas to rise from their slumber!</span>")
-	notify_ghosts("\The <b>[caller]</b> is calling for the burrowed larvas to wake up!", enter_link = "join_larva=1", enter_text = "Join as Larva", source = caller, action = NOTIFY_JOIN_AS_LARVA)
+
+	var/burrowed = caller.hive.stored_larva
+	var/total = 0
+	for(var/i in 1 to burrowed)
+		var/mob/M = get_alien_candidate()
+		if(!M)
+			continue
+
+		var/list/possible_mothers = list()
+		SEND_SIGNAL(src, COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, possible_mothers) //List variable passed by reference, and hopefully populated.

```

Comment:
Signals are supposed to be unique when sent.
Make a `COMSIG_HIVE_XENO_MOTHER_CALLOFTHEBURROWED` or similar.

Reply:
It's not that simple since it registers  `RegisterSignal(caller.hive, list(COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, COMSIG_HIVE_XENO_MOTHER_CHECK), .proc/is_burrowed_larva_host)` and adding more signals would mean that I would have to register even more in various places. Tbh this shouldn't have been a signal but a proc instead.

Reply:
In the Crash PR this is being expanded to include not only xenos, but also structures (the silos). That expansion possibility is on of the reasons why signals are good. Changing it to a proc would require to have a list of the listeners and trigger it for each. That's what signals do.

But yes, you need to add the define for the new signal and register it along with the others. for it to work.

Reply:
Sure I can do it if you dont mind doing it this way.

Reply:
It's fine.
You can also make it call a generic (hive) proc more things use that sends the signal as well. Then you can reduce the 3 signals to one.